apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -228,14 +229,20 @@ export class UpsertWorkflowUseCase {
   }
 
   private buildBaseStepEntity(
-    step: StepDto | StepUpdateDto,
+    step: StepCreateDto | StepUpdateDto,
     foundPersistedStep?: NotificationStepEntity
   ): NotificationStep {
+    const controlDto = step.controlSchema ? { schema: step.controlSchema } : null;
+    const outputDto = step.outputSchema ? { schema: step.outputSchema } : null;
+    const outputFallback = { schema: stepTypeToDefaultDashboardControlSchema[step.type].schema };
```

Comment:
To align framework and dashboard workflows, I added a fallback for the output schema from controls schema. 
In framework workflows, the output schema is always present, while in dashboard workflows, the controls currently match the outputs one-to-one. If this changes in the future, this section will need to be updated.

Reply:
I don't know if that's true. Controls don't much outputs 100% in the Dashboard.

Reply:
you are correct, we need to map the email one. i forgot about it, thanks!

self-note: check if there are any other misalignments besides email.